playbooks/roles/buildbot-master/files/master.cfg

```diff
@@ -0,0 +1,201 @@
+# --master option)
+c['protocols'] = {'pb': {'port': 9989}}
+
+####### CHANGESOURCES
+
+# the 'change_source' setting tells the buildmaster how it should find out
+# about source code changes.  Here we point to the buildbot clone of pyflakes.
+
+c['change_source'] = []
+c['change_source'].append(changes.GitPoller(
+        'git://github.com/buildbot/pyflakes.git',
+        workdir='gitpoller-workdir', branch='master',
+        pollinterval=300))
+
+####### SCHEDULERS
+
+# Configure the Schedulers, which decide how to react to incoming changes.  In this
+# case, just kick off a 'runtests' build
+
+c['schedulers'] = []
+c['schedulers'].append(schedulers.SingleBranchScheduler(
+                            name="all",
+                            change_filter=util.ChangeFilter(branch='master'),
+                            treeStableTimer=None,
+                            builderNames=["runtests"]))
+c['schedulers'].append(schedulers.ForceScheduler(
+                            name="force",
+                            builderNames=["runtests"]))
+c['schedulers'].append(schedulers.ForceScheduler(
+                            name="force_configuration_tests",
+                            builderNames=["run_configuration_tests"]))
+
+####### BUILDERS
+
+# The 'builders' list defines the Builders, which tell Buildbot how to perform a build:
+# what steps, and which workers can execute them.  Note that any particular build will
+# only take place on one worker.
+
+factory = util.BuildFactory()
+# check out the source
+factory.addStep(steps.Git(repourl='git://github.com/buildbot/pyflakes.git', mode='incremental'))
+# run the tests (note that this will require that 'trial' is installed)
+factory.addStep(steps.ShellCommand(command=["trial", "pyflakes"]))
+
+# Factory for testing configuration repo
+config_test_factory = util.BuildFactory()
+config_test_factory.addStep(steps.Git(
+    name='clone git repo',
+    repourl='git://github.com/Stanford-Online/configuration.git',
+    mode='incremental',
+))
+config_test_factory.addStep(steps.ShellSequence(
+    name='install dependencies via apt-get',
+    commands=[
+        util.ShellArg(command=['sudo', 'apt-get', 'update']),
+        util.ShellArg(command=['sudo', 'apt-get', 'install', '-y', 'nodejs']),
+        util.ShellArg(command=['sudo', 'apt-get', 'install', '-y', 'python-demjson']),
+    ],
+))
+config_test_factory.addStep(steps.ShellSequence(
+    name='install dependencies via pip',
+    commands=[
+        util.ShellArg(command=['sudo', 'pip', 'install', '--allow-all-external', '-r', 'requirements.txt']),
+        util.ShellArg(command=['sudo', 'pip', 'install', '--allow-all-external', 'demjson']),
+    ],
+))
+config_test_factory.addStep(steps.ShellCommand(
+    name='test YAML files',
+    command=(
+        'for yml in $(find . -name "*.yml"); do\n'
+          'python -c "import sys,yaml; yaml.load(open(sys.argv[1]))" $yml >/dev/null\n'
+          'if [ $? -ne 0 ]; then\n'
```

Comment:
I had to replace the `[[` and `]]` in the `if` statement with `[` and `]`, but it seems to work. I think maybe I'm using a different vagrant box than normal?